Lib/test/test_csv.py

```diff
@@ -189,6 +189,29 @@ def test_write_escape(self):
         self._write_test(['a',1,'p,q'], 'a,1,p\\,q',
                          escapechar='\\', quoting = csv.QUOTE_NONE)
 
+    def test_write_escape_escapechar(self):
+        tests = [
+            # fields, expected output, quoting
+            (['a/', 1], '"a//",1', csv.QUOTE_MINIMAL),
```

Comment:
It is indeed hard to understand what this test is actually checking, so we should improve it.

I suggest using a different format: (values, expected NONE, expected MINIMAL, expected ALL). IMO this would make the set of tested inputs, and the effects of the different quoting methods, clearer.

Also, consider adding a comment about the rationale for each test case, e.g. "string ending with escape char", "string with no escape chars", "single escape char only".